### PR TITLE
Add docs for `EsploraAsyncExt` and make doctests runnable

### PR DESCRIPTION
**Pull request by evanlinjin**
### Description

Add docs for `EsploraAsyncExt` and make various doctests runnable.

### CODE REVIEW

File: crates/esplora/src/async_ext.rs [hunk 1]
```diff
@@ -12,6 +12,12 @@ use futures::stream::{FuturesOrdered, TryStreamExt};
 
 use crate::map_confirmation_time;
 
+/// Trait to extend [`esplora_client::AsyncClient`] functionality.
+///
+/// This is the async version of [`EsploraExt`]. Refer to [crate-level documentation] for more.
```
Comment (rajarshimaitra) [inline crates/esplora/src/async_ext.rs]: I couldn't find any more info on the crate-level documentation. Maybe something more will be added later?

**Review by rajarshimaitra — COMMENTED** [review]
tACK 136a0b130cd8fda167e099823ec775635b5f7cce

File: crates/esplora/src/async_ext.rs (hunk 1 above)
Comment (evanlinjin) [reply crates/esplora/src/async_ext.rs]: Woops you are right. Maybe "refer to crate level documentation for usage" will be better?

Comment (evanlinjin) [reply crates/esplora/src/async_ext.rs]: Fixed.

Comment (rajarshimaitra) [reply crates/esplora/src/async_ext.rs]: Umm.. It also doesn't have any usage docs too. Can't we just point to the `keychain_tracker_esplora` crate for usage? That's where they will find how to use this trait.

Comment (evanlinjin) [reply crates/esplora/src/async_ext.rs]: How about now?

**Review by rajarshimaitra — APPROVED** [review]
ACK 877b6587875e75407567d9080f486eba2cb08460

**Comment by ChidiChuks** [comment]
@evanlinjin Great work on this PR! Considering the increasing adoption of Miniscript for expressive and secure policy construction in Bitcoin, would it be worth referencing or aligning this implementation with ongoing Miniscript support? This could help ensure compatibility and future-proofing, especially for more complex script use cases.

**Pull request merged**